documentation/src/main/asciidoc/user_guide/streams.adoc

```diff
@@ -412,12 +412,20 @@ method on the `CacheStream`.  This value will default to the `chunkSize` configu
 Unfortunately this value is a tradeoff with memory usage vs performance vs at least once and your
 mileage may vary.
 
-*Using `iterator` with a replication cache*
+*Using `iterator` with cache replication*
+
+For replicated caches, {brandname} performs the `iterator` or `spliterator` terminal operations locally on the node, which optimizes performance as remote iterations are more resource intensive.
+
+[NOTE]
+====
+//Documents ISPN-9332
+When using cache stores that are both `shared` and have `write-behind` enabled, {brandname} performs iterations remotely to ensure consistency.
+====
+
+When write behind is enabled, the write modification is stored on the primary owner. REPL local iteration can read from non owned data, thus causing an inconsistency.
+
+Thus distributed streams should always go remote when not all segments are primarily owned on a given node when write behind is enabled.
```

Comment:
This line seems a bit out of place now and doesn't explicitly mention that it is a distributed cache that it is talking about. I think clarifying in the above section would be better.